### ISSUE TITLE
Relay 10: Error when compiling new directives

### ISSUE THREAD

**Issue by wadamek65** [relay-tools/relay-compiler-language-typescript#195]
### Description
After updating relay packages versions to v10, I am unable to use new directives `appendEdge`, `prependEdge` and `deleteRecord`.

### Usage
For the following query:
```ts
  const [commitCreateProfile] = useMutation<
    CreateCreateProfileMutation
  >(graphql`
    mutation CreateCreateProfileMutation($input: CreateProfileInput!) {
      createProfile(input: $input) {
        result @appendEdge(connections: ["Normal_profiles"]){
          node {
            id
            category
            subcategories
            status
          }
        }
        error {
          message
        }
      }
    }
  `);
```
`relay-compiler` throws an error:
```
C:\code\work\kinderway\frontend>yarn relay
yarn run v1.18.0
$ yarn run relay-compiler --schema ../schema.graphql --src ./src/ --language typescript
$ C:\code\work\kinderway\node_modules\.bin\relay-compiler --schema ../schema.graphql --src ./src/ --language typescript
Writing ts
ERROR:
Unknown directive 'appendEdge'.
error Command failed with exit code 100.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 100.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Package versions
```
"react-relay": "^10.0.0",
"relay-runtime": "^10.0.0",
"babel-plugin-relay": "^10.0.0",
"relay-compiler": "^10.0.0",
"relay-compiler-language-typescript": "^12.0.3",
"relay-config": "^10.0.0",
"relay-test-utils": "^10.0.0",
```

`yarn why output`:
```
C:\code\work\kinderway>yarn why relay-compiler
yarn why v1.18.0
[1/4] Why do we have the module "relay-compiler"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "relay-compiler@9.1.0"
info Has been hoisted to "relay-compiler"
info Reasons this module exists
   - "workspace-aggregator-83504b82-9adb-45fa-83e4-d3d172b23434" depends on it
   - Hoisted from "_project_#@graphql-codegen#typescript#@graphql-codegen#visitor-plugin-common#@graphql-tools#relay-operation-optimizer#relay-co
mpiler"
   - Hoisted from "_project_#backend#graphql-tools#@graphql-tools#relay-operation-optimizer#relay-compiler"
info Disk size without dependencies: "3.55MB"
info Disk size with unique dependencies: "12.95MB"
info Disk size with transitive dependencies: "3.55MB"
info Number of shared dependencies: 142
=> Found "frontend#relay-compiler@10.0.0"
info This module exists because "_project_#frontend" depends on it.
Done in 1.93s.
```

**Comment by wadamek65** [comment]
Update: Working on fixing this right now, will try to submit a draft PR tomorrow.

**Comment by wadamek65** [comment]
After a long time debugging I've figured out why this happens. Here is an explanation:

I had multiple packages installed that depended on graphql-tools which in turn installed relay-optimizer and it uses relay-compiler v9. yarn insisted on using that binary instead of the v10.0.0 one which resulted in that error. After symlinking it to a correct binary it now works. Still no code completion for the directives but at least the compilation step works properly.

So this issue is partly unrelated to `relay-compiler-language-typescript`. Still no code completion for the new directives but I'm not sure which package is responsible for adding these directives into schema. I think this issue can be closed.

**Comment by alloy** [comment]
:rocket: Issue was released in v13.0.1 :rocket:
